pkg/operations/kubernetesupgrade/upgradecluster.go

```diff
@@ -127,6 +127,9 @@ func (uc *UpgradeCluster) getClusterNodeStatus(az armhelpers.AKSEngineClient, re
 			return err
 		}
 		for _, vmScaleSet := range vmScaleSetPage.Values() {
+			if !utils.IsVMSSUpgradeable(uc.AgentPoolsToUpgrade, *vmScaleSet.Name) {
```

Comment:
can we have this method in the operations method itself?

Reply:
I want to keep the name parsing logic out of the upgrade code. Is there a way we can do this with the common name parsing logic in the utils and only upgrade related part here  ?

Reply:
Got it. In that case, can we have the method that extracts the agentPoolName (in `armhelpers` package)and then the utility method (perhaps in helpers package. It could even be a private local method.) which checks if the string exists in the map or not.

Thoughts on this?

Reply:
Reasoning for this is to have small and focused methods.

Reply:
Thanks Tariq, will give that try and get back in next iteration.

Reply:
Thanks for your patience with this PR :)